src/main.c

```diff
@@ -307,5 +317,88 @@ __attribute__((section(".boot"))) int main(void) {
         END_TRY;
     }
     app_exit();
+}
+
+static void start_app_from_lib(void) {
+    G_called_from_swap = true;
+    G_swap_response_ready = false;
+    UX_INIT();
+    io_seproxyhal_init();
+    nv_app_state_init();
+    USB_power(0);
+    USB_power(1);
+#ifdef HAVE_BLE
+    // Erase globals that may inherit values from exchange
+    MEMCLEAR(G_io_asynch_ux_callback);
+    // grab the current plane mode setting
+    G_io_app.plane_mode = os_setting_get(OS_SETTING_PLANEMODE, NULL, 0);
+    BLE_power(0, NULL);
+    BLE_power(1, "Nano X");
+#endif  // HAVE_BLE
+    app_main();
+}
+
+static void library_main_helper(libargs_t *args) {
+    check_api_level(CX_COMPAT_APILEVEL);
+    switch (args->command) {
+        case CHECK_ADDRESS:
+            // ensure result is zero if an exception is thrown
+            args->check_address->result = 0;
+            args->check_address->result = handle_check_address(args->check_address);
```

Comment:
Would it be clearer if `args->check_address->result` was a `bool` instead of an `int`?